src/util/error.ts

```diff
@@ -44,6 +47,12 @@ export enum Code {
 
   // ErrInvalidArgument is returned when the argument is invalid.
   ErrInvalidArgument = 'ErrInvalidArgument',
+
+  // ErrOperationNotReady is returned when another operation needs to be completed first.
+  ErrOperationNotReady = 'ErrOperationNotReady',
+
+  // ErrOperationNotPermitted is returned when the operation is not permitted.
+  ErrOperationNotPermitted = 'ErrOperationNotPermitted',
```

Comment:
The document you referred to was very helpful. Based on your suggestions, I have updated the error codes as follows:

```javascript
// ErrNotInitialized is returned when the required initialization has not been completed.
ErrNotInitialized = 'ErrNotInitialized',

// ErrNotReady is returned when the execution of the following actions is not ready.
ErrNotReady = 'ErrNotReady',

// ErrRefused is returned when the execution is rejected.
ErrRefused = 'ErrRefused',
```

Instead of grouping all errors under `ErrNotReady`, I've decided to seperate them to `ErrNotInitialized` and `ErrNotReady` for clarity. Additionally, `ErrNotPermitted` could be misinterpreted as a permission related issue, so I've decided to change error name to `ErrRefused`.